### PR TITLE
Add BEL graph writer

### LINKED ISSUE #29: Add BEL graph writer
The ecosystem of Biological Expression Language (BEL) -aware software is growing, especially in the knowledge graph embedding space (as shown by BioKEEN, which OpenBioLink references)

It would be nice to make and maintain an export of the OpenBioLink graph in BEL. I'd be happy to make a PR that takes care of this after we merge #28.

### PR DESCRIPTION
**Pull request by cthoyt**
Closes #29

This pull request adds support for exporting of positive and negative graphs in the Biological Expression Language (BEL) format for usage with [PyBEL](https://github.com/pybel/pybel) and other BEL-aware applications. It exports to both the BEL Script format (which is canonically what everyone can use in any software) and a pre-compiled JSON format for direct, fast use with PyBEL (preferred).

It also updates the README with some link for BEL and PyBEL so users might have some ideas what they can do with the resulting output.

### CODE REVIEW

File: src/openbiolink/graph_creation/graphCreation.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
 FORMATS = {
     "TSV": GraphTSVWriter,
     "RDF-N3": GraphRDFWriter,
+    "PICKLE": GraphPickleWriter,
```
Comment (cthoyt) [inline src/openbiolink/graph_creation/graphCreation.py]: Forgot this in the last PR, now the CLI can pick pickle as well

File: src/openbiolink/graph_creation/graph_writer/bel_writer.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    _type_to_adder = {
+        EdgeType.DRUG_BINDINH_GENE: BELGraph.add_directly_inhibits,
+        EdgeType.GENE_GENE: BELGraph.add_association,
+        EdgeType.GENE_OVEREXPRESSED_ANATOMY: BELGraph.add_positive_correlation,
+        EdgeType.GENE_GO: BELGraph.add_part_of,
+        EdgeType.GENE_INHIBITION_GENE: BELGraph.add_inhibits,
+        EdgeType.GENE_EXPRESSED_ANATOMY: BELGraph.add_association,
+        EdgeType.DRUG_ACTIVATION_GENE: BELGraph.add_activates,
+        EdgeType.DRUG_PREDBIND_GENE: None,
```
Comment (cthoyt) [inline src/openbiolink/graph_creation/graph_writer/bel_writer.py]: Several of these edge types have a `None` because I'm not sure about the biological interpretation. could you explain what prebind is, what PTM the `GENE_PTMOD_GENE` means, what `GENE_REACTION_GENE` means, and what the catalysis interaction is?

File: src/openbiolink/node.py [hunk 3]
```diff
@@ -3,10 +3,11 @@
 
 
 class Node:
-    def __init__(self, id: str, type: NodeType, namespace: Namespace, name=""):
+    def __init__(self, id: str, type: NodeType, namespace: Namespace, name=None):
```
Comment (cthoyt) [inline src/openbiolink/node.py]: Not sure what the impact of this change is, but it seems bad edge case to use an empty string if there's no data. Why don't all nodes have required names, though? This is something that would be really helpful for looking at the network and its edges

Comment (nomisto) [reply src/openbiolink/node.py]: I don't think the variable name is used ever, as far as I can tell. You mean a clear name like https://hpo.jax.org/app/browse/term/HP:0000118 "Phenotypic abnormality"?

Comment (cthoyt) [reply src/openbiolink/node.py]: yeah, like having the label to go with each node is really powerful. The label for each node should be available in the source ontologies and terminologies

File: src/openbiolink/graph_creation/graph_writer/bel_writer.py (hunk 2 above)
Comment (nomisto) [reply src/openbiolink/graph_creation/graph_writer/bel_writer.py]: "Predbind" means "predicted to bind" and "ptmod" for "post-translational modifications" as in 4. of https://github.com/OpenBioLink/OpenBioLink/issues/22#issuecomment-582174838.
@matthias-samwald @annabreit Maybe you can help with this?

File: src/openbiolink/node.py (hunk 3 above)
Comment (nomisto) [reply src/openbiolink/node.py]: I have to look into that, but this would indeed be a nice feature.

**Comment by cthoyt** [comment]
I've released [PyBEL v0.14.4](https://github.com/pybel/pybel/releases/tag/v0.14.4) which has all of the updates needed for this PR, but we're still waiting for discussion about PTMOD (which one? always phosphorylation? BEL doesn't have a way to write a generic PTM, so this will have to lose granularity), about REACTION, and about CATALYSIS

**Comment by nomisto** [comment]
I don't know if this will help you, but these edges were extracted from the STRING db. (9606.protein.actions.v11.0.txt.gz in https://string-db.org/cgi/download.pl?species_text=Homo+sapiens) 

Unfortunately the doc of the file, does not give a detailed explaination of the action modes: https://string-db.org/help/faq/#what-does-the-columns-in-proteinsactions-file-mean

I've tried checking what PTM is meant by ptmod manually and I am by no means an expert, but I think it is not only phosphorylation.
F.e. One edge in the graph is https://string-db.org/cgi/network.pl?taskId=s9XIuCkkHRQm. A click on the connections -> button "show" besides Post-translational modification, reveals that this edges is a "... Methylation of NDUFS2 by NDUFAF7"

**Comment by cthoyt** [comment]
@nomisto thanks for the links. I guess they're using text mining for PTMs since the kind is only in free text. That means that we can use it to infer the "regulates" edge in BEL, but no more.

Now that I'm aware that you're including predicted binding information in your network, I would suggest discontinuing import of that information for any link prediction task, as it would conflate the previously used prediction method with whatever link prediction task is being evaluated.

From what it said, I think reaction is also a superclass of directed binding interactions, so I'll make a BEL export based on that. Will mark for review soon!

**Comment by nomisto** [comment]
Looks good!

> Now that I'm aware that you're including predicted binding information in your network, I would suggest discontinuing import of that information for any link prediction task, as it would conflate the previously used prediction method with whatever link prediction task is being evaluated.

That's a good point. STRING, for example, creates these predictions through:
- NLP
- Transferred from other species
- Computed from genomic features

The last point could be problematic. 
I'll talk with @matthias-samwald about this.

**Comment by cthoyt** [comment]
Perhaps a good idea to make a new issue. It won't have any effect on the BEL export, so I wouldn't say it blocks this PR
